### PR TITLE
Migrate to use v0.14 API

### LINKED ISSUE #5: Migrate to Fluentd v0.14 API

### PR DESCRIPTION
**Pull request by cosmo0920**
Hi, thanks for paying attention to use v0.14 Plugin API.

I've tried to migrate to use v0.14 Output Plugin API.
This PR contains major update change.
Could you bump up minor version if releasing new version of gem?
And please refer http://docs.fluentd.org/v0.14/articles/plugin-update-from-v12 before release new version.

Thanks,

Will fix #5.

### CODE REVIEW

**Comment by daichirata** [comment]
@cosmo0920 Thank you for pull request and a good job! :tada:

I added some comments so please check it.

File: lib/fluent/plugin/out_gcs.rb [hunk 1]
```diff
@@ -126,19 +148,29 @@ def hex_random(chunk)
       Digest::MD5.hexdigest(chunk.unique_id)[0...@hex_random_length]
     end
 
-    def format_path(chunk)
-      now = Time.strptime(chunk.key, @time_slice_format)
+    def format_path(time_slice_format, time_slice)
```
Comment (cosmo0920) [inline lib/fluent/plugin/out_gcs.rb]: Oops! :scream:

File: lib/fluent/plugin/out_gcs.rb [hunk 2]
```diff
@@ -126,19 +148,29 @@ def hex_random(chunk)
...
+                     Time.at(metadata.timekey).utc.strftime(time_slice_format)
+                   end
       tags = {
         "%{file_extension}" => @object_creator.file_extension,
         "%{hex_random}" => hex_random(chunk),
         "%{hostname}" => Socket.gethostname,
         "%{index}" => i,
-        "%{path}" => format_path(chunk),
-        "%{time_slice}" => chunk.key,
+        "%{path}" => format_path(time_slice_format, time_slice),
```
Comment (cosmo0920) [inline lib/fluent/plugin/out_gcs.rb]: Hmm, you are right.
We can use `extract_placeholders` here like as fluent-plugin-s3.

**Comment by cosmo0920** [comment]
@daichirata I've tried to fix your comments.
Could you take a look, again?

File: lib/fluent/plugin/out_gcs.rb [hunk 3]
```diff
@@ -126,22 +149,26 @@ def hex_random(chunk)
...
     end
 
-    def format_path(chunk)
-      now = Time.strptime(chunk.key, @time_slice_format)
-      (@localtime ? now : now.utc).strftime(@path)
-    end
-
     def generate_path(chunk, i = 0, prev = nil)
+      metadata = chunk.metadata
+      time_slice_format = @configured_time_slice_format || timekey_to_timeformat(@buffer_config['timekey'])
```
Comment (daichirata) [inline lib/fluent/plugin/out_gcs.rb]: Please delete this line.

File: README.md [hunk 4]
```diff
@@ -11,6 +11,35 @@ gem install fluent-plugin-gcs
...
+<match pattern>
+  @type gcs
+
+  project YOUR_PROJECT
+  keyfile YOUR_KEYFILE_PATH
+  bucket YOUR_GCS_BUCKET_NAME
+  object_key_format %{path}%{time_slice}_%{index}.%{file_extension}
+  path logs/${tag}/%Y/%m/%d/
+
+  # if you want to use ${tag} or %Y/%m/%d/ like syntax in path / s3_object_key_format,
```
Comment (daichirata) [inline README.md]: Please correct `s3_object_key_format` to `object_key_format`.

**Comment by cosmo0920** [comment]
Fixed.

**Comment by daichirata** [comment]
@cosmo0920 Thank you for fixing for fluentd v0.14! :confetti_ball::tada::sparkles:
I will give you 100 github points :gift: :octocat: 

> Could you bump up minor version if releasing new version of gem?

OK, I will release the next version on v0.5.0.beta1.

**Pull request merged**